### ISSUE TITLE
BulkCopy inserts binary data incorrectly

### ISSUE THREAD

**Issue by MaceWindu** [mysql-net/MySqlConnector#816]
During [implementation](https://github.com/linq2db/linq2db/pull/2215) of BulkCopy support in linq2db we found out that binary types inserted incorrectly using BulkCopy API (other types seems to work).

Looks like inserted data inserted like `binary -> hex string -> binary`:
```
insert: byte[3] { 1, 2, 3 }
in db: byte[6] { 48, 49, 48, 50, 48, 51 } // "010203"
```

**Comment by bgrainger** [comment]
This is covered in the documentation: https://mysqlconnector.net/api/mysql-bulk-copy/

> Columns containing binary data must be mapped using an expression that uses the `UNHEX` function.

The underlying reason is that `MySqlBulkCopy` just uses the `LOAD DATA` command, which accepts string data, and I don't think there's a way to safely escape arbitrary binary data (e.g., that contains newlines.

**Comment by MaceWindu** [comment]
Thanks, will see what we can do on our side

**Comment by bgrainger** [comment]
This is definitely a usability problem for the API, though. At the (small) cost of reading the destination table schema, MySqlConnector could detect destination columns that are binary, and automatically set up the `UNHEX` expression. This would also be helpful for `BIT` columns, that need a custom expression, too.

**Comment by bgrainger** [comment]
This is fixed in [0.67.0](https://github.com/mysql-net/MySqlConnector/releases/tag/0.67.0).
